cmd/podman/containers/restore.go

```diff
@@ -99,26 +99,40 @@ func restore(cmd *cobra.Command, args []string) error {
 		return fmt.Errorf("restoring a container requires root")
 	}
 
-	// Find out if this is an image
-	inspectOpts := entities.InspectOptions{}
-	imgData, _, err := registry.ImageEngine().Inspect(context.Background(), args, inspectOpts)
-	if err != nil {
-		return err
+	// Check if the container exists (#15055)
+	exists := &entities.BoolReport{Value: false}
+	for _, ctr := range args {
+		exists, err := registry.ContainerEngine().ContainerExists(registry.GetContext(), ctr, entities.ContainerExistsOptions{})
```

Comment:
Should this be?
 		exists, err = registry.ContainerEngine().ContainerExists(registry.GetContext(), ctr, entities.ContainerExistsOptions{})

So that exists will be still set when this block exits?

Reply:
Thank you for reviewing.
`exists` is only set in this block.
I'll fix it.